pkg/storage/chunk/chunk.go

```diff
@@ -72,9 +80,14 @@ const BatchSize = 12
 // small, and passed by value.
 type Batch struct {
 	Timestamps [BatchSize]int64
-	Values     [BatchSize]float64
-	Index      int
-	Length     int
+
+	ValueTypes           chunkenc.ValueType
```

Comment:
I don't fully understand this. It's plural but holds a single type. Can't a batch contain a mix of types?

Reply:
Current semantics does not allow multiple types, I think this is a case of bad naming. I suggest to change the name and not rewrite the semantics, wdyt?

Reply:
Sure, I agree. The other comments I left assume the value of a batch is always 1.

Reply:
ended up reverting it and only deal with Floats in the Batch